src/coreclr/vm/gc_unwind_x86.inl

```diff
@@ -3824,3 +3824,350 @@ bool EnumGcRefsX86(PREGDISPLAY     pContext,
+        }
+
+        if (isZero)
+        {
+            // we always encode zero in short form
+            m_val = 0;
+        }
+    }
+    else
+    {
+        m_val = (m_vals.m_chunks[0] << 1) & arg.m_val;
+    }
+}
+
+void ptrArgTP::doBigOrAssign(const ptrArgTP& arg)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+        SUPPORTS_DAC;
+    } CONTRACTL_END;
+
+    //
+    // Change to Big representation
+    //
+    toBig();
+
+    if (arg.isBig())
+    {
+        unsigned myLen  = m_vals.GetLength();
+        unsigned argLen = arg.m_vals.GetLength();
+
+        if (myLen < argLen)
+        {
+            // expand our length to match argLen and zero init
+            memset(m_vals.m_chunks + myLen, 0, sizeof(ChunkType) * (argLen - myLen));
+            m_vals.SetLength(argLen);
+            myLen = argLen;
+        }
+
+        for(unsigned i = 0; ((i < myLen) && (i < argLen)); i++)
+        {
+            m_vals.m_chunks[i] |= arg.m_vals.m_chunks[i];
+        }
+    }
+    else
+    {
+        m_vals.m_chunks[0] |= arg.smallBits();
+    }
+}
+
+void ptrArgTP::doBigDiffAssign(const ptrArgTP& arg)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+        SUPPORTS_DAC;
+    } CONTRACTL_END;
+
+    //
+    // Change to Big representation
+    //
+    toBig();
+
+    unsigned myLen  = m_vals.GetLength();
+    unsigned argLen = arg.m_vals.GetLength();
+    bool     isZero = true;                    // until proven otherwise
+
+    for (unsigned i = 0; (i < myLen); i++)
+    {
+        ChunkType nextChunk = m_vals.m_chunks[i];
+        if (i < argLen)
+        {
+            nextChunk &= ~arg.m_vals.m_chunks[i];
+            m_vals.m_chunks[i] = nextChunk;
+        }
+        else if (i == 0)
+        {
+            nextChunk &= ~arg.smallBits();
+            m_vals.m_chunks[i] = nextChunk;
+        }
+
+        if (nextChunk != 0)
+            isZero = false;
+    }
+
+    if (isZero)
+    {
+        // we always encode zero in short form
+        m_val = 0;
+    }
+}
+
+BOOL ptrArgTP::doBigEquals(const ptrArgTP& arg) const
+{
+    CONTRACT(BOOL)
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        SUPPORTS_DAC;
+    }
+    CONTRACT_END;
+
+    unsigned myLen  = m_vals.GetLength();
+    unsigned argLen = arg.m_vals.GetLength();
+    unsigned maxLen = (myLen >= argLen) ? myLen : argLen;
+
+    for (unsigned i=0; (i < maxLen); i++)
+    {
+        ChunkType myVal  = 0;
+        ChunkType argVal = 0;
+
+        if (i < myLen)
+            myVal = m_vals.m_chunks[i];
+
+        if (i < argLen)
+            argVal = arg.m_vals.m_chunks[i];
+
+        if (i == 0)
+        {
+            if (myLen == 0)
+                myVal = smallBits();
+            if (argLen == 0)
+                argVal = arg.smallBits();
+        }
+
+        if (myVal != argVal)
+            RETURN false;
+    }
+    RETURN true;
+}
+
+BOOL ptrArgTP::doBigIntersect(const ptrArgTP& arg) const
+{
+    CONTRACT(BOOL)
```

Comment:
All contracts in this and the other file can be unified to `LIMITED_METHOD_DAC_CONTRACT` that can simplify the dummy macro definitions for NAOT a bit.